app/services/hardware/hardware.ts

```diff
@@ -0,0 +1,86 @@
+import { mutation, StatefulService } from '../stateful-service';
+import * as obs from '../../../obs-api';
+import uuid from 'uuid/v4';
+
+export enum EDeviceType {
+  audioInput = 'audioInput',
+  audioOutput = 'audioInput',
+  videoInput = 'audioInput'
+}
+
+export interface IDevice {
+  id: string;
+  type: EDeviceType;
+  description: string;
+}
+
+export interface IHardwareServiceState {
+  devices: IDevice[];
+}
+
+export class HardwareService extends StatefulService<IHardwareServiceState> {
+
+  static initialState: IHardwareServiceState = {
+    devices: []
+  };
+
+  init() {
+    this.SET_DEVICES(this.fetchDevices());
+  }
+
+  getDevices() {
+    return this.state.devices;
+  }
+
+  getDevice(id: string) {
+    return this.state.devices.find(device => device.id == id);
+  }
+
+  getDeviceByName(name: string) {
+    return this.state.devices.find(device => device.description == name);
```

Comment:
i'm not sure why we're filing these under `device.description` when they're referred to as `name` everywhere else